cmd/kubeadm/app/discovery/token/token.go

```diff
@@ -226,14 +226,27 @@ func fetchKubeConfigWithTimeout(apiEndpoint string, discoveryTimeout time.Durati
 	}
 }
 
-// parsePEMCert decodes a PEM-formatted certificate and returns it as an x509.Certificate
-func parsePEMCert(certData []byte) (*x509.Certificate, error) {
-	pemBlock, trailingData := pem.Decode(certData)
-	if pemBlock == nil {
-		return nil, errors.New("invalid PEM data")
-	}
-	if len(trailingData) != 0 {
-		return nil, errors.New("trailing data after first PEM block")
+// parsePEMCerts decodes a PEM-formatted certificates into a slice of x509.Certificates
+func parsePEMCerts(certData []byte) ([]*x509.Certificate, error) {
```

Comment:
please modify this function to:
```
// parsePEMCerts decodes PEM-formatted certificates into a slice of x509.Certificates
func parsePEMCerts(certData []byte) ([]*x509.Certificate, error) {
	var certificates []*x509.Certificate
	var pemBlock *pem.Block

	for {
		pemBlock, certData = pem.Decode(certData)
		if pemBlock == nil {
			return nil, errors.New("invalid PEM data")
		}

		cert, err := x509.ParseCertificate(pemBlock.Bytes)
		if err != nil {
			return nil, errors.Wrap(err, "unable to parse certificate")
		}
		certificates = append(certificates, cert)

		if len(certData) == 0 {
			break
		}
	}
	return certificates, nil
}
```

note the `a PEM-` -> `PEM` change in the function comment.

Reply:
Thanks! Fixed